Encode any 3 character ASCII currency code (#106)
ripple/ripple-binary-codec: README.md

  * If other fields (in the future) must to support X-addresses with tags, this library will need to be updated.
  * When decoding rippled binary, the output will always output classic address + tag, with no X-addresses. X-address support only applies when encoding to binary.

> #### Encoding Currencies

Reviewer: ```suggestion
#### Encoding Currency Codes
```

>   * ripple-binary-codec allows for any 3 character ASCII string to be encoded into a [currency code](https://xrpl.org/currency-formats.html#currency-codes).

Reviewer: ```suggestion
  * The standard format for currency codes is a three-letter string such as `USD`. This is intended for use with ISO 4217 Currency Codes.
  * Currency codes must be exactly 3 ASCII characters in length and there are [a few other rules](https://xrpl.org/currency-formats.html#currency-codes). ripple-binary-codec allows any 3-character ASCII string to be encoded as a currency code, although rippled may enforce tighter restrictions.
```

>   * ripple-binary-codec will decode currencies into that are valid ISO 4217 codes.

Reviewer: ```suggestion
  * When _decoding_, if a currency code is three uppercase letters or numbers (`/^[A-Z0-9]{3}$/`), then it will be decoded into that string. For example,`0000000000000000000000004142430000000000` decodes as `ABC`.
```

>   * If a currency is not a valid ISO 4217 code, it will return a 160-bit hex-string.

Reviewer: ```suggestion
  * When decoding, if a currency code is does not match the regex, then it is not considered to be an ISO 4217 or pseudo-ISO currency. ripple-binary-codec will return a 160-bit hex-string (40 hex characters). For example, `0000000000000000000000006142430000000000` (`aBC`) decodes as `0000000000000000000000006142430000000000` because it contains a lowercase letter.
```
